src/arrayMethodJoin.js

```diff
@@ -5,8 +5,30 @@
  *
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function() {
-    // write code here
+  [].__proto__.join2 = function(str) {
+    let string = '';
+    let separator = str;
+    if (separator === undefined) {
+      separator = ',';
+    }
+    if (separator === null) {
+      separator = 'null';
+    }
+    if (this.length === 0) {
+      return string;
+    } else {
```

Comment:
you don\`t need `else` after `return`. Moreover, you can miss it because with `length === 0` loop  will work 0 times